lib/beaker-hostgenerator/data.rb

```diff
@@ -29,6 +56,873 @@ def pe_upgrade_family
       'm' => 'master',
     }
 
+    BASE_CONFIG = {
+      'HOSTS' => {},
+      'CONFIG' => {
+        'nfs_server' => 'none',
+        'consoleport' => 443,
+      }
+    }
+
+    def base_host_config
+      {
+        'pe_dir' => pe_dir(pe_version, pe_family),
+        'pe_ver' => pe_version,
+        'pe_upgrade_dir' => pe_dir(pe_upgrade_version, pe_upgrade_family),
+        'pe_upgrade_ver' => pe_upgrade_version,
+      }
+    end
+
+    def osinfo
+      {
```

Comment:
@rick Have you looked into how rspec-puppet-facts work? It sounds just like the model your proposing.

https://github.com/mcanevet/rspec-puppet-facts which at it's core, depends on facterdb - a gem that contains json blobs of facter output by facter version and os.

https://github.com/camptocamp/facterdb/tree/master/facts/3.1
https://github.com/camptocamp/facterdb/blob/master/lib/facterdb.rb